CardsCashCasino/Manager/TexasHoldEmManager.cs

```diff
@@ -665,74 +772,249 @@ private void HandleBettingPhase()
         }
 
+        /// <summary>
+        /// handles the initial stage of betting before any community cards appear
+        /// </summary>
         private void HandlePreflop()
         {
+            if (_players.OnePlayerLeft())
+            {
+                return;
+            }
+            int playerIndex;
             // Set the current bet to the big blind.
             _currentBet = _bigBlindBet;
-            
-            // Set the player index to the player with the small blind
-            int playerIndex = _currentSmallBlind;
-            
-            // iterate through the players starting with the small blind. and handle their actions.
-            for (int i = 0; i < _playerHands.Count; i++)
+
+            // Set the player index to the player to the left of the big blind
+            playerIndex = _players.GetPreflopStartingBettor();
+
+            // iterate through the players starting with the player to the left of the big blind. and handle their actions.
+            //loop will terminate and betting round will end once conditions are met
+            while (!_players.AdvanceRound() && !_players.OnePlayerLeft())
+            {
+                if (!_players.AdvanceToRoundConclusion())
+                {
+                    _currentPhase = Phase.INIT;
+                    RoundConclusion();
+                    return;
+                }
+                // If the player is the user, set the user playing flag to true.
+                if (playerIndex == 0)
+                    _userPlaying = true;
+
+                // If the player is an AI player, set the AI playing flag to true.
+                else
+                    _userPlaying = false;
+                if (_players.IsActivePlayer(playerIndex))
+                {
+                    // Handle the player's action.
+                    HandlePlayerAction(playerIndex);
+                }
+
+                // Increment the player index.
+                playerIndex = (playerIndex + 1) % _playerHands.Count;
+            }
+
+            //finalize bets for the round and add them to the pots
+            _potManager.AddFoldedBetsToPot(_players.PackageFoldedBets());
+            _potManager.AddToPot(_currentBet, _players.PackageBets());
+
+            //reset the bets for the next round to 0
+            _players.ResetBets();
+
+        }
+
+        /// <summary>
+        /// rules, flow and exceptions for the second round of betting that follows the flop
+        /// </summary>
+        private void HandleFlop()
+        {
+            if (_players.OnePlayerLeft())
+            {
+                return;
+            }
+            int playerIndex;
+            _currentBet = 0;
+            // Set the player index to the player to the left of the big blind
+            playerIndex = _players.GetStartingBettorIndex();
+
+            // Set the player index to the small blind or the player closest to the small blind
+            while (!_players.AdvanceRound())
```

Comment:
The same question as above comes here. We are nesting a while loop within the main Update loop, which we cannot do. That will just stall the game. Refactor this and the above to be within the Update loop.